### ISSUE TITLE
Helper creation in 5.2alpha

### ISSUE THREAD

**Issue by Goatlips** [petergridge/Irrigation-V5#6]
Hi,
Just planning on setting up irrigation v5,   On paper it looks perfect for my application.  With the changes to the structure I am finding it hard to determine what i have to create in Yaml and what i have to create in the UI and what helpers that i need to generate and in the upcoming releases are these done automatically.  Is the latest alfa release fully UI based...  If it is happy to do some new user beta testing.  I dont want to invest hours into yaml only to have you release a new update

**Comment by petergridge** [comment]
Hi @goatlips

Version 5 is fully UI based. The current Alpha is predominantly upgrading the Zone Group feature to be configured in the config flow. Adding a few bells and whistles, and a lot of code clean up and an improved version (I hope everyone agrees) of the custom card. It’s amazing how with every release I manage to find better (hopefully) ways to do things.

You will need to create helpers; input_number, input_select, input_boolean objects to support the features you want to use but this can all be done via the UI now so no YAML.

As a bare minimum for a simple four zone irrigation system where all zones run on the same schedule:
•	1x Input_datetime for the program start time
•	1x input_boolean for the toggling of the config in the custom card
•	1x input_select for the frequency of the program, assuming they all run every time a program is started, you can define the frequency for each zone, I create one per zone as this my the favoured way to turn off a zone as well. You would need to create an input_boolean to support the disable zone feature anyway so two for the price of one from the custom card perspective.
•	4x input_number for the duration of each zone, you could use one if they all have the same duration, but that is lazy 😊

After that you could add more helpers/sensors to support ECO, adjustments, rain sensors etc. to extend the implementation.

I will be releasing a new alpha soon, I still have work to do on the custom card. 

I am always after testers, this is a beast to test now, I have added a simulate service that writes a lot of information to the log that I can check that the calculations etc are accurate but there a so many permutations for the zone configuration now that there are always some surprises. 

I hope this helps with your decision.

Cheers
Pete

**Comment by petergridge** [comment]
Hi @Goatlips ,

**alpha 2 is available!** :boom: 

This is a big change under the covers I will release this as version 5.2.0 I think it warrants a major release status. There has been a lot of changes under the covers, every time I add more features I find better ways to structure the code. So I am sure you will find issues with this version.

I would really appreciate any support with testing of this version.

Enjoy
Pete

**Custom Control**
Expanded config Flow to run of a menu for new programs.
* Migrate groups from the UI to Config Flow. Existing groups are automatically migrated to config flow
groups can then be added or deleted in the config flow.
* If a zone is deleted or is changed to reference a different switch the affect groups will be deleted
and you will need to add a new group. 
* This change has allowed better controls and validation of the running of the program and facilitates improvements to the Custom Card

* Interlock - you can now turn off the interlock to let multiple programs run simultaneously. This is designed for large implementations where multiple independent irrigations are required. WARNING: be cautious where multiple programs referencing the same zone.

* Warnings raised in the log when a program is stopped by another program or service call.

* Additional attributes have been added to the event data for the start of a zone

*  Program/card reacts when a zone switch is turned off manually the zone turns off and the next zone in a running program will run.

* Refactored the watering timer to support:
	* program level total runtime 
	* interactive changes to adjustment values, water/wait/repeat cycle configuration while a program is running
	
	
**Custom Card** 

* Zones grouped into a single section. Assuming grouped zones have the same attributes. i.e. use the same input objects

* For large implementations where a single long card is not practical you can now provide a list of zones to show in a card,  allowing multiple cards to be displayed for a single program

```yaml
type: custom:irrigation-card-test
program: switch.test_irrigation
show_program: true
entities:
  - switch.dummy_1
  - switch.dummy_2
```
* Make the card more compact

* Removed any boiler plate text to improve presentation in languages other than English.

* display relative time since a program was run on the card, eg 10 minutes, 1 day...

* Zone stop option provided on the card when a zone is running. If you stop a zone the remaining zones of a program will continue

**Comment by Goatlips** [comment]
Thanks Pete,  I will setup alpha2 over the next couple of days and do some testing.  My system will certainly test your processes.  I will flick any feedback and bugs

**Comment by petergridge** [comment]
Hi @Goatlips,

I have released V5.2.0-beta, it is stable and waiting for feedback from the early adopters that they have no issues with their implementations.

**Comment by petergridge** [comment]
Closing this issue, conversion continues in Issue #13
